dll/win32/shell32/CShellLink.cpp

```diff
@@ -2209,11 +2209,16 @@ HRESULT CShellLink::SetTargetFromPIDLOrPath(LPCITEMIDLIST pidl, LPCWSTR pszFile)
             StringCchCopyW(szPath, _countof(szPath), pszFile);
             PathResolveW(szPath, NULL, PRF_TRYPROGRAMEXTENSIONS);
 
-            pidlNew = SHSimpleIDListFromPathW(szPath);
-            /******************************************************/
-            /* Question: Why this line is needed only for files?? */
-            hr = (*szPath ? S_OK : E_INVALIDARG); // S_FALSE
-            /******************************************************/
+            if (PathIsFileSpecW(szPath))
+            {
+                hr = E_INVALIDARG;
+                szPath[0] = 0;
+            }
+            else
+            {
+                hr = S_OK;
+                pidlNew = SHSimpleIDListFromPathW(szPath);
```

Comment:
Nice, but now I wonder: What if `SHSimpleIDListFromPathW()` failed, by returning `NULL`? Should we fail?

Reply:
I think not. Adding the following code:
```c
                if (pidlNew == NULL)
                    hr = E_INVALIDARG;
```
will increase one failure:
![VirtualBox_ReactOS_24_06_2021_10_59_39](https://user-images.githubusercontent.com/2107452/123190916-5f7ebd80-d4db-11eb-88ff-e7ba9f671b37.png)

Reply:
thanks! Maybe this needs to be mentioned in a comment (that we don't fail on purpose even if `SHSimpleIDListFromPathW` returns NULL).

Reply:
For example?

Reply:
Let me think about one; I will post below when I get an idea.

Reply:
```c
// We don't fail on purpose even if SHSimpleIDListFromPathW returns NULL).
```

Reply:
something like that; maybe add that this behaviour has been verified with tests.

Reply:
Comments added.